tls/s2n_encrypted_extensions.c

```diff
@@ -27,26 +27,39 @@
 
 /**
   * Specified in https://tools.ietf.org/html/rfc8446#section-4.3.1
-  * 
+  *
   * In all handshakes, the server MUST send the EncryptedExtensions
-  * message immediately after the ServerHello message.  
+  * message immediately after the ServerHello message.
   *
   * The EncryptedExtensions message contains extensions that can be
   * protected, i.e., any which are not needed to establish the
   * cryptographic context but which are not associated with individual
-  * certificates. 
+  * certificates.
   **/
 
 static int s2n_server_encrypted_extensions_parse(struct s2n_connection *conn, struct s2n_blob *extensions);
 
+int s2n_encrypted_extensions_send_size(struct s2n_connection *conn)
+{
+    /* Calculate size of encrypted extensions. */
+    int total_size = 0;
+
+    total_size += s2n_server_extensions_server_name_send_size(conn);
+    total_size += s2n_server_extensions_max_fragment_length_send_size(conn);
+    total_size += s2n_server_extensions_alpn_send_size(conn);
+
+    return total_size;
+}
+
 int s2n_encrypted_extensions_send(struct s2n_connection *conn)
 {
+    S2N_ERROR_IF(conn->actual_protocol_version != S2N_TLS13, S2N_ERR_BAD_MESSAGE);
     struct s2n_stuffer *out = &conn->handshake.io;
 
-    /* Calculate size of encrypted extensions. For minimal TLS 1.3, this is 0
-     * as we are sending an empty EE message
-     */
-    uint16_t total_size = 0;
+    const int total_size = s2n_encrypted_extensions_send_size(conn);
+
+    GUARD(total_size);
```

Comment:
This could be the first use of `inclusive_range_check` outside the tests :) just a nit.

This is a good check. Can this also be done for the regular server hello (which also covers HRR, which can have cookies which can be arbitrarily large).

Reply:
cool 👍